File: src/app/Models/TaxStatus.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use App\Traits\Auditable;

class TaxStatus extends Model
{
    use HasFactory, SoftDeletes, Auditable;

    protected $fillable = [
        'code',
        'name',
        'description',
        'is_active',
    ];

    protected $casts = [
        'is_active' => 'boolean',
    ];

    /**
     * Relación: Un tax status puede tener muchos proveedores
     */
    public function providers()
    {
        return $this->hasMany(Provider::class);
    }
}

File: backend-dotnet/src/Ikhana.Domain/Entities/TaxStatus.cs
namespace Ikhana.Domain.Entities;

public class TaxStatus : BaseEntity, ISoftDelete, IAuditable
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime? DeletedAt { get; set; }

    public ICollection<Provider> Providers { get; set; } = new List<Provider>();
}
